toucan_connectors/connection_manager.py

```diff
@@ -0,0 +1,184 @@
+import logging
+import threading
+import time
+import types
+from enum import Enum
+from typing import Dict, Optional
+
+logger = logging.getLogger(__name__)
+
+
+class Status(Enum):
+    READY = 'ready'
+    IN_PROGRESS = 'in progress'
+    CLOSE_IN_PROGRESS = 'is_closing'
+
+
+class ConnectionObject:

```

Comment:
naming : `ConnectionObject` because of the Object type I guess? but Object is really wide term in python, can we narrow this down ?